02-peer-protocol.md

```diff
@@ -894,40 +916,43 @@ any message), they are independent of requirements here.
 ### Requirements
 
 A node MUST handle continuing a previous channel on a new encrypted
-transport.  On disconnection, a node MAY forget nodes which have not
-sent or received an `accept_channel` message, and MAY forget nodes
-which have not sent `funding_locked` after a reasonable timeout.
+transport.
+
+On disconnection, the funder MUST remember the channel for
+reconnection if it has broadcast the funding transaction, otherwise it
+MUST NOT.
+
+On disconnection, the non-funding node MUST remember the channel for
+reconnection if it has sent the `funding_signed` message, otherwise
+it MUST NOT.
 
 On disconnection, a node MUST reverse any uncommitted updates sent by
 the other side (ie. all messages beginning with `update_` for which no
 `commitment_signed` has been received).  Note that a node MAY have
 already use the `payment_preimage` value from the `update_fulfill_htlc`,
 so the effects of `update_fulfill_htlc` is not completely reversed.
 
-On reconnection, a node MUST retransmit old messages which may not
+On reconnection, if a channel is in an error state, the node SHOULD
+retransmit the error packet and ignore any other packets for that
+channel, or if the channel has entered closing negotiation, the node
+MUST retransmit the last `closing_signed`.
+
+Otherwise, a node MUST retransmit old messages after `funding_signed` which may not
 have been received, and MUST NOT retransmit old messages which have
 been explicitly or implicitly acknowledged.  The following table
 lists the acknowledgment conditions for each message:
 
-* `open_channel`: acknowledged by `accept_channel`.
-* `accept_channel`: acknowledged by `funding_created`.
-* `funding_created`: acknowledged by `funding_signed`.
-* `funding_signed`: acknowledged by `funding_locked`.
 * `funding_locked`: acknowledged by `update_` messages, `commitment_signed`, `revoke_and_ack` or `shutdown` messages.
 * `update_` messages: acknowledged by `revoke_and_ack`.
 * `commitment_signed`: acknowledged by `revoke_and_ack`.
 * `revoke_and_ack`: acknowledged by `commitment_signed` or `closing_signed`
 * `shutdown`: acknowledged by `closing_signed`.
 
-The last `closing_signed` (if any) must always be retransmitted, as there
-is no explicit acknowledgment.
-
-Before retransmitting `commitment_signed`, the node MUST send
-appropriate `update_` messages (the other node will have forgotten
-them, as required above).
+On reconnection, the node MUST retransmit all `update_` messages, and
```

Comment:
Shall we say that on reconnect both parties have to roll back any changes since the last committed state, and then replay them? Instead of allowing states to diverge and then trying to reconcile it by discarding duplicates. Would allow for a brief inconsistency if nodes replay different updates, but they may only act on them once they were committed to anyway. May be a bit simpler to grok, since it aligns with the transactional model.